charts/payment-integration/templates/deployment.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: {{ template "payment.fullname" . }}
+  labels:
+    app: {{ template "payment.name" . }}
+    chart: {{ template "payment.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  replicas: {{ default 2 .Values.replicaCount }}
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app: {{ template "payment.name" . }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+{{ toYaml .Values.rollingUpdateStrategy | indent 6 }}
+  minReadySeconds: 5
+  template:
+    metadata:
+      labels:
+        app: {{ template "payment.name" . }}
+      annotations:
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: '{{ .Values.service.internalPort }}'
+    spec:
+      containers:
+      - name: {{ .Values.containerName }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        ports:
+        - name: {{ .Values.portName }}
+          containerPort: {{ .Values.service.internalPort }}
+        # Inject (non sensitive) environment variables from the `Values` config.
+        env:
+        {{- range $key, $val := .Values.nonSensitiveEnvs }}
+        - name: {{ $key }}
+          value: {{ $val | quote }}
+        {{- end}}
+        # Inject (sensitive) environment variables from `Values` config, referring
+        # them from the related Secret object. This is an extra measure of security
+        # to avoid having credentials displayed in plain text in the template file.
+        # https://kubernetes.io/docs/concepts/configuration/secret/
+        {{- $chartName := .Chart.Name -}}
```

Comment:
variable `chartName` is currently not used here anymore. please have a look at how the existing charts are rendering the env variables (using `include` instead of `template` function) https://github.com/commercetools/k8s-charts/blob/0a89d4688d7cc855a7581b6a46278c594786ed25/charts/cronjob/templates/cronjob.yaml#L38